### PR TITLE
Add hgcalv16 modifier and corresponding era

### PR DESCRIPTION
**Pull request by srimanob**
#### PR description:
This PR is to add a new modifier for HGCAL V16, and the corresponding era. Currently, HGCal V16 (C17) is used in D86 and D88. It needs new RecHits calibration, introduced in https://github.com/cms-sw/cmssw/pull/36728. We need to split the modifier/era of D86 and D88 from C11, in order to keep D77 (default now) working as before.

#### PR validation:
This PR itself should do nothing. To see the effect of new RecHit calibration, it should be triggerred the test with https://github.com/cms-sw/cmssw/pull/36728

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
No need of backporting.

### CODE REVIEW

**Comment by srimanob** [comment]
@cmsbuild please test

File: Configuration/Geometry/python/dict2026Geometry.py [hunk 1]
```diff
@@ -1144,7 +1144,7 @@
             'from Geometry.EcalMapping.EcalMapping_cfi import *',
             'from Geometry.EcalMapping.EcalMappingRecord_cfi import *',
         ],
-        "era" : "phase2_ecal, phase2_hcal, phase2_hgcal, hcalHardcodeConditions, phase2_hgcalV10, phase2_hgcalV11, phase2_hgcalV12, phase2_hfnose",
+        "era" : "phase2_ecal, phase2_hcal, phase2_hgcal, hcalHardcodeConditions, phase2_hgcalV10, phase2_hgcalV11, phase2_hgcalV16, phase2_hfnose",
```
[the remark this answers is not in the document]
Comment (srimanob) [reply Configuration/Geometry/python/dict2026Geometry.py:1147]: @kpedro88 Do you have an idea what is the result of this era line? I think it does not change anything at the moment. Thanks in advance.

[the remark this answers is not in the document]
Comment (kpedro88) [reply Configuration/Geometry/python/dict2026Geometry.py:1147]: @srimanob indeed, right now it is just printed as a recommendation when a user runs the geometry script. The actual modifiers for a workflow have to be implemented manually.

**Comment by srimanob** [comment]
As described, no change is expected from this PR itself. The test with HGCal V16 rechit calibration is done in https://github.com/cms-sw/cmssw/pull/36728#issuecomment-1017035817

**Comment by srimanob** [comment]
+Upgrade

The failure in PR test doest not relate to this PR. PR test alone shows no change as expected. Testing with https://github.com/cms-sw/cmssw/pull/36728 shows expected behaviour, https://github.com/cms-sw/cmssw/pull/36728#issuecomment-1017179820

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
Kindly ping @cms-sw/pdmv-l2

**Comment by kskovpen** [comment]
+1

**Comment by perrotta** [comment]
+oprations
- Successfully tested together with #36728, see https://github.com/cms-sw/cmssw/pull/36728#issuecomment-1024904105

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**